### DISCUSSION TITLE
Is tangling files possible?

### DISCUSSION THREAD

**Discussion by ghost** [nvim-orgmode/orgmode#275] [Q&A]
Is it possible to export the code segments of a .org file to its own file? For example, can I export Python snippets throughout a file called README.org to a file called config.py?
Update: I found a Python script called [org-babel-tangle.py](https://github.com/OrgTangle/org-babel-tangle.py) that does tangling from the command line so I don't need to use Emacs. I'm still wondering if there's any way to implement it as a Neovim plugin, though.

**Comment by andreadev-it** [comment]
I'm looking for the same thing (but for lua)!
I could also write a plugin to do this, but I don't want to reinvent the wheel if not necessary 🙂
In that case, what would be the best syntax for it that is supported by treesitter? I'm asking for opinions of course, nothing too deep. I was thinking of using a `:TANGLE:` property under the headings, but for the whole document it doesn't look like it could work (as it doesn't work the original `#+property: header-args :tangle [file]` by the way)

**Comment by TravonteD** [comment]
You can create a user command that calls that script on the current file. Here's an example.
```lua
vim.api.nvim_create_user_command('OrgTangle', function()
  local cmd = string.format('terminal org-babel-tangle.py %s', vim.fn.expand('%'))
  vim.cmd(cmd)
end, {buffer = true})
```

**Reply by refaelsh** [reply]
Great function. Using it :-)

**Comment by refaelsh** [comment]
> Is it possible to export the code segments of a .org file to its own file? For example, can I export Python snippets throughout a file called README.org to a file called config.py? Update: I found a Python script called [org-babel-tangle.py](https://github.com/OrgTangle/org-babel-tangle.py) that does tangling from the command line so I don't need to use Emacs. I'm still wondering if there's any way to implement it as a Neovim plugin, though.

This is cool. I tried it, and it works. Thank you for posting this :-)

P.S.
May I suggest please that this functionality should be made part of the plugin?

**Comment by refaelsh** [comment]
> Is it possible to export the code segments of a .org file to its own file? For example, can I export Python snippets throughout a file called README.org to a file called config.py? Update: I found a Python script called [org-babel-tangle.py](https://github.com/OrgTangle/org-babel-tangle.py) that does tangling from the command line so I don't need to use Emacs. I'm still wondering if there's any way to implement it as a Neovim plugin, though.

After some playing with `org-babel-tangle.py`, I see that you need to repeat this `:tangle /path/to/file` for every `#+BEGIN_SRC` block. Tedious. Bummer.

**Reply by andreadev-it** [reply]
You might want to take a look at a section of my neovim config. I've implemented a similar code for tangling files in my org configuration:

https://github.com/andreadev-it/nvim-personal-config/blob/main/org-config.org

Just thought it might be useful

**Reply by refaelsh** [reply]
Yes. I am looking at it now. Thank you very much :-)

**Reply by jgollenz** [reply]
@andreadev-it feel free to open up a PR for that, if you want ;) tangling is part of Emacs orgmode and should therefore be part of this repo as well :+1:

**Reply by andreadev-it** [reply]
Hi @jgollenz ,
Currently, my code does not reflect perfectly the way that org mode handle this situation on emacs, since it seems that some of those options are not supported by treesitter (or at least were not supported when I last checked). Do you believe it's still doable to implement this, or would it be better to wait (or work) and get a full implementation of the original syntax?

**Reply by jgollenz** [reply]
In general, this project tries to be a faithful port where possible. However, in my opinion, supporting a feature like this partly (ideally with a plan how to support it fully in the future) is better than not supporting it at all. Please open up a draft PR so we can discuss the issues in detail :+1:

**Reply by andreadev-it** [reply]
@jgollenz Done, both for this and for the multipurpose enter key discussion 🙂
